install/infra/terraform/gke/main.tf

```diff
@@ -0,0 +1,191 @@
+terraform {
+  required_version = ">= 1.0.3"
+}
+
+terraform {
+  backend "gcs" {
+    bucket = "gitpod-gke"
+    prefix = "tf-state"
+  }
+}
+
+provider "google" {
+  project     = var.project
+  credentials = var.credentials
+  region      = var.region
+  zone        = var.zone
+}
+
+resource "google_compute_network" "vpc" {
+  name                    = "vpc-${var.name}"
+  auto_create_subnetworks = "false"
+}
+
+resource "google_compute_subnetwork" "subnet" {
+  name          = "subnet-${var.name}"
+  region        = var.region
+  network       = google_compute_network.vpc.name
+  ip_cidr_range = "10.255.0.0/16"
+
+  secondary_ip_range {
+    range_name    = "cluster-secondary-ip-range"
+    ip_cidr_range = "10.0.0.0/12"
+  }
+
+  secondary_ip_range {
+    range_name    = "services-secondary-ip-range"
+    ip_cidr_range = "10.64.0.0/12"
+  }
+}
+
+resource "google_container_cluster" "gitpod-cluster" {
+  name     = "c${var.name}"
+  location = "${var.zone}" == "" ? "${var.region}" : "${var.region}-${var.zone}"
+
+  cluster_autoscaling {
+    enabled = true
+
+    resource_limits {
+      resource_type = "cpu"
+      minimum       = 2
+      maximum       = 8
+    }
+
+    resource_limits {
+      resource_type = "memory"
+      minimum       = 4
+      maximum       = 64
+    }
+  }
+
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So we create the smallest possible default
+  # node pool and immediately delete it.
+  remove_default_node_pool = true
```

Comment:
This is a great point. The solution to re-use the default node as `service` node is something I thought too. But I couldn't see an option at least in the [container cluster terraform resource](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster) to set labels only to the default node(the labels being `meta` and `ide`). If I set the `label` field, it would apply the same label to every nodepool. For the testing purposes, I imagine this is probably okay. Not sure if this is the best course of action.

Reply:
Turns out, I couldn't have been more wrong. That option only applied the labels to the default node. Making this change! Thanks for the suggestion, Mo!